core/modules.rs

```diff
@@ -390,7 +321,7 @@ pub trait ModuleLoader {
   /// dynamic imports altogether.
   fn load(
     &self,
-    module_specifier: &ModuleSpecifier,
+    module_specifier: ModuleSpecifier,
```

Comment:
Why change these types here? It appears to me that each call will incur a clone of `ModuleSpecifier`

Reply:
Ah yeah. It's not a clone, but rather a stack copy. 

From the benchmarks it appeared that compiler optimizations get rid of the overhead of this entirely, however I think I can revert these changes as we can probably make `ModuleSpecifier` a `FastString` as well to remove any chance that a future compiler change may make this slower. The benchmarks didn't show any slowdown, but looking at the real size of the underlying `Url` struct does make me less confident this will always be the case.

Reply:
Interesting observation. I think it makes sense, but I'm a bit hesitant because if we use `FastString` here we lose the enforcement that this field is a valid `ModuleSpecifier` (which is an alias to `Url`) - that's the assumption we make everywhere we use ES modules that the specifier is a valid URL. Can you think of any idea to enforce it while using `FastString`?

Reply:
Right now, not really. There are a lot more invariants around the `ModuleSpecifier`s than `ModuleName`s and code, so either we'd have to parse back into a `Url` when we actually do loading from the filesystem/network (potentially delaying errors). I definitely think we can punt this until later down the line and think about it more.

Reply:
(reverted these changes for now -- we can re-address later on)